### PR TITLE
Remove JavaScript From Stored XSS Labels

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
This change removes a vulnerability related to stored xss on translated labels.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
***lurch: attach W-039440*

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HsHEQA0" target="_blank">W-039440</a>

**Comment by npsp-reedestockton** [comment]
@RadSFDeveloper @force2b Ready for review

File: src/classes/UTIL_HtmlOutput_TEST.cls [hunk 1]
```diff
@@ -122,4 +122,20 @@ public with sharing class UTIL_HtmlOutput_TEST {
...
         System.assertNotEquals(html, actualHtml);
     }
+
+    /*******************************************************************************************************
+    * @description Verifies that non-whitelisted javascript event code is stripped from url
+    */
+    @isTest
+    private static void shouldStripJavaScriptFromUrl() {
+        String html = '<a href="https://powerofus.force.com/NPSP_Documentation" onanyjavascriptevent="alert(\'gotcha\')">';
+        string cleanHtml = '<a href="https://powerofus.force.com/NPSP_Documentation" >';
```
Comment (force2b) [inline src/classes/UTIL_HtmlOutput_TEST.cls]:
```suggestion
        String cleanHtml = '<a href="https://powerofus.force.com/NPSP_Documentation" >';
```

File: src/classes/UTIL_HtmlOutput_TEST.cls [hunk 2]
```diff
@@ -122,4 +122,20 @@ public with sharing class UTIL_HtmlOutput_TEST {
...
+        UTIL_HtmlOutput_CTRL controller = new UTIL_HtmlOutput_CTRL();
+
+        controller.unsafeHtml = html;
+        controller.hasUrl = true;
+
+        System.assertEquals(cleanHtml, controller.getSafeHtml());
```
Comment (force2b) [inline src/classes/UTIL_HtmlOutput_TEST.cls]: Would this also work if the bad html is one of following:

- `'<a href=\'https://powerofus.force.com/NPSP_Documentation\' onanyjavascriptevent=\'alert("gotcha")\'>';`
- `'<a href="javascript:alert(\'gotcha\')">'`

Comment (npsp-reedestockton) [reply src/classes/UTIL_HtmlOutput_TEST.cls]: This is handled by the existing HtmlOutput code.  It essentially just becomes a text string that is never invoked.

**Comment by npsp-reedestockton** [comment]
@force2b Requested change has been made.

File: src/classes/UTIL_HtmlOutput_TEST.cls (hunk 2 above)
Comment (npsp-reedestockton) [reply src/classes/UTIL_HtmlOutput_TEST.cls]: The first example is stripped by the new code.

Comment (force2b) [reply src/classes/UTIL_HtmlOutput_TEST.cls]: Confirmed. Thanks!!

**Comment by rponti-sforg** [comment]
QE testing is complete for this PR and corresponding WI.  Tests have passed.

For further testing details, see:
https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HsHEQA0/view
https://salesforce.quip.com/oc7ZAIb9Db4d

Once PR Feature Test - Namespaced test completes and passes, will merge PR.  Once PR post merge builds pass, will set corresponding WI status to 'Closed'.

**Comment by rponti-sforg** [comment]
@codexmarcel Looks like Feature Test - Namespaced has possibly stalled.  Are you able to give it a kick?

**Comment by force2b** [comment]
@rponti-sforg I restarted the Namespaced build.

**Comment by rponti-sforg** [comment]
> @rponti-sforg I restarted the Namespaced build.

Thank you @force2b.  Appreciate it.  Merged!

**Pull request merged**